### ISSUE TITLE
Installation with Python2 error

### ISSUE THREAD

**Issue by navakk9** [learningequality/kolibri#1836]
## Summary

Couldn't install latest 0.5.-Beta2.pex file with Python2, broke with the below error message. Worked with Python3.

## System information

Version: 0.5.0-Beta2.pex
OS: Mac

## Traceback or relevant snippet from server.log or browser console
![screen shot 2017-07-11 at 9 00 12 pm](https://user-images.githubusercontent.com/26124065/28102442-106431e0-6684-11e7-9cfd-4e462c01259a.png)

## How to reproduce

1. Follow installation steps with Python2

**Comment by navakk9** [comment]
@aronasorman

**Comment by aronasorman** [comment]
On the six library:

![image](https://user-images.githubusercontent.com/191955/28130031-711cf6a0-66ea-11e7-9aa1-5a55ce31999d.png)

**Comment by aronasorman** [comment]
I thought six made that import available regardless of what version we're using. Looks like we might need to use the classic try catch import style.

**Comment by aronasorman** [comment]
So the issue is that @navakk9 has three pythons: The 2.7 from the python website, 3.5 from homebrew, and 2.7 that's built in to the Mac (located at `/usr/bin/python`). Running `python kolibri.pex` or `python3 kolibri.pex` works, because it uses the first two, which seems to be doing the right thing and adding the`six` library we want.

The issue is when we do `./kolibri.pex`, which calls the builtin python located at `/usr/bin/python`. Instead of using the `six` module bundled in `django.dist`, it's using another `six` module found in `/Library`.

Basically, it boils down to incompatible libraries. The built-in python may have additional safeguards and restrictions regarding path modification that's making us use an old version of `six`. The fix here might be to add `six` into BBQ itself (six is implemented in just one file), then have BBQ use that directly, instead of depending on an installed vesrion.

**Comment by rtibbles** [comment]
Fixed in #1831

**Comment by benjaoming** [comment]
> I thought six made that import available regardless of what version we're using. Looks like we might need to use the classic try catch import style.

That fine. IMO the important thing is to keep compatibility issues together, for instance in `kolibri.utils.compat` - the point being to avoid repeating the same patterns, and to have them easily swapped out or deleted when we go "py3-only".

Btw. *don't* use `future`, that was the library that broke our py2+3 dist and I removed it from django-q's requirements.
